broadinstitute/single_cell_portal_core
# Validate cell names are unique for Metadata and Cluster files (SCP-3767)

This adds client-side validation to check that the cell names in a Metadata or Cluster file are all unique. If there is a duplicated cell name it is displayed to the user with a request to fix the duplicates.

To test:
- pull this branch
- go to the upload wizard in a study you can add files to
- try to upload the file that is included in this PR: `test/js/lib/validate-file-content.test.js`
- you should see an error message indicating the duplicate cell names

Error message is added to the list of errors if a file has multiple things wrong:
![Screen Shot 2021-11-22 at 12 00 43 PM](https://user-images.githubusercontent.com/54322292/142918807-92e4e0ba-d992-4053-b7f6-8a140664418f.png)

Error message is shown for a file with duplicated cell names
![Screen Shot 2021-11-22 at 11 58 27 AM](https://user-images.githubusercontent.com/54322292/142918909-ede5b666-204a-408d-bc6a-3b4196f18a81.png)
